lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java

```diff
@@ -143,12 +143,13 @@ public APIGatewayProxyResponseEvent handleRequest(
             } else {
                 Map<String, List<String>> authParameters =
                         getAuthParamsAsMap(clientOAuthSessionItem);
-
-                var validationResult =
-                        authRequestValidator.validateRequest(authParameters, input.getHeaders());
+                Map<String, String> params = new HashMap<>();
+                params.put(IPV_SESSION_ID_HEADER_KEY, ipvSessionId);
+                params.put(CLIENT_SESSION_ID_HEADER_KEY, clientSessionId);
+                var validationResult = authRequestValidator.validateRequest(authParameters, params);
                 if (!validationResult.isValid()) {
-                    return ApiGatewayResponseGenerator.proxyJsonResponse(
-                            HttpStatus.SC_BAD_REQUEST, validationResult.getError());
+                    return new JourneyErrorResponse(
+                            null, HttpStatus.SC_BAD_REQUEST, validationResult.getError());
```

Comment:
I think, we need to sure what set value is null or JOURNEY_ERROR_PATH for journey here